main/mainworker.cpp

```diff
@@ -11741,7 +11755,7 @@ bool MainWorker::SwitchLight(const uint64_t idx, const std::string &switchcmd, c
     if (_log.isTraceEnabled()) _log.Log(LOG_TRACE,"MAIN SwitchLight idx:%d cmd:%s lvl:%d " ,(long)idx,switchcmd.c_str(),level );
 	std::vector<std::vector<std::string> > result;
 	result = m_sql.safe_query(
-		"SELECT HardwareID,DeviceID,Unit,Type,SubType,SwitchType,AddjValue2,nValue,sValue,Name,Options FROM DeviceStatus WHERE (ID == %" PRIu64 ")",
+		"SELECT HardwareID,DeviceID,Unit,Type,SubType,SwitchType,AddjValue2,nValue,sValue,Name,Options,LastLevel FROM DeviceStatus WHERE (ID == %" PRIu64 ")",
```

Comment:
I dont see where you use the 'LastLevel' from the sql query.. maybe this can be removed here (could save some cpu/disk cycles)

Reply:
LastLevel is used here: `level = atoi(sd[11].c_str());`, level will be 0 if it was not passed either by JSON API or MQTT API, then stored LastLevel will be used instead.
A cleaner solution would be to modify the code such that missing value is indicated with an illegal level, for example -1, but it requires much more changes to the code.
By the way, is it unnecessary to overwrite level with LastLevel only if  switchtype != STYPE_Selector?

Reply:
Ok, now i understand, this is used in MainWorker::SwitchLightInt
But then, you also have to push something in the WebServer.cpp , line 4627 is calling SwitchLightInt also)